pandas/tests/frame/test_query_eval.py

```diff
@@ -35,6 +35,12 @@ def skip_if_no_pandas_parser(parser):
 
 
 class TestCompat:
+    def test_query_non_boolean_raise_error(self):
```

Comment:
Can you add tests that use EAs (e.g. the nullable boolean dtype) and where the `res.ndim` from above is 2.